src/Sentry/Extensibility/SentryStackTraceFactory.cs

```diff
@@ -68,47 +69,41 @@ internal SentryStackTrace Create(StackTrace stackTrace, bool isCurrentStackTrace
                 stacktrace.Frames.Add(frame);
             }
 
-            return stacktrace.Frames.Count == 0
-                ? null
-                : stacktrace;
+            return stacktrace.Frames.Count != 0
+                ? stacktrace
+                : null;
         }
 
         /// <summary>
         /// Creates an enumerator of <see cref="SentryStackFrame"/> from a <see cref="StackTrace"/>.
         /// </summary>
-        /// <param name="stackTrace"></param>
-        /// <param name="isCurrentStackTrace"></param>
-        /// <returns></returns>
         internal IEnumerable<SentryStackFrame> CreateFrames(StackTrace stackTrace, bool isCurrentStackTrace)
         {
-            var frames = stackTrace?.GetFrames();
+            var frames = stackTrace.GetFrames();
             if (frames == null)
             {
+                // TODO: this probably can never happen
```

Comment:
Some environments don't bring stack traces like some Unity targets
Could be it returns 0 size array though, can't recall.

I'd say the TODO here is not helpful. We can discuss and remove the code if it's certain that it's never null.
```suggestion
```